variables.tf

```diff
@@ -62,64 +57,72 @@ variable "agent_name" {
   nullable    = false
 }
 
-variable "agent_resource_group_name" {
+variable "agent_resource_group_id" {
   type        = string
-  description = "The resource group name for the schematics agent."
+  description = "The resource group ID of the schematics resource group."
   nullable    = false
 }
 
 variable "schematics_location" {
   type        = string
-  description = "The location to create the Schematics workspace. Allowed values are `us-south`, `us-east`, `eu-gb`, `eu-de`, `ca-tor`, `ca-mon`, `eu-fr2`."
+  description = "List of locations supported by IBM Cloud Schematics service. Allowed values are `us-south`, `us-east`, `eu-gb`, `eu-de`."
   default     = "us-south"
   validation {
-    condition     = contains(["us-south", "us-east", "eu-gb", "eu-de", "ca-tor", "ca-mon", "eu-fr2"], var.schematics_location)
-    error_message = "Allowed values for `schematics_location` are \"us-south\", \"us-east\", \"eu-gb\", \"ca-tor\", \"ca-mon\", \"eu-fr2\" or \"eu-de\"."
+    condition     = contains(["us-south", "us-east", "eu-gb", "eu-de"], var.schematics_location)
+    error_message = "Allowed values for `schematics_location` are \"us-south\", \"us-east\", \"eu-gb\" or \"eu-de\"."
   }
 }
 
-variable "use_schematics_private_endpoint" {
-  type        = bool
-  description = "Set to `true` to use IBM Cloud Schematics private endpoints. Requires the runtime to have access to the IBM Cloud private network."
-  default     = false
-}
-
 variable "agent_version" {
   type        = string
   description = "The schematics agent version. More info: https://cloud.ibm.com/docs/schematics?topic=schematics-update-agent-overview&interface=ui#agent_version-releases"
   nullable    = false
   default     = "1.5.0"
+}
 
-  validation {
-    condition     = contains(["1.5.0", "1.4.0", "1.3.1", "1.3.0", "1.2.0", "1.1.1", "1.1.0", "1.0.0", ], var.agent_version)
-    error_message = "Agent version provided is not supported."
-  }
+# Agent Policy
+variable "create_agent_policy" {
+  type        = bool
+  description = "Flag when set to True creates schematics agent policy."
+  default     = false
```

Comment:
can we add some variable validation here. If this is set to true, certain other inputs are now required